tests/java/InferBearTest.py

```diff
@@ -20,8 +22,66 @@ class InferBad {
 }
 """
 
+good_objc_file = """
+#import <Foundation/Foundation.h>
+
+@interface Hello: NSObject
+@property NSString* s;
+@end
+
+@implementation Hello
+NSString* m() {
+    Hello* hello = nil;
+    return hello.s;
+}
+@end
+"""
+
+bad_objc_file = """
+#import <Foundation/Foundation.h>
+
+@interface Hello: NSObject
+@property NSString* s;
+@end
+
+@implementation Hello
+NSString* m() {
+    Hello* hello = nil;
+    return hello->_s;
+}
+@end
+"""
+
+good_c_file = """
+void test() {
+  int *s = NULL;
+  if (s != NULL) {
+    *s = 42;
+  }
+}
+"""
+
+bad_c_file = """
+#include <stdlib.h>
+
+void test() {
+  int *s = NULL;
+  *s = 42;
+}
+"""
+
+
+InferBearJavaTest = verify_local_bear(
+    InferBear, valid_files=(good_java_file,), invalid_files=(bad_java_file,),
+    tempfile_kwargs={"suffix": ".java"}, settings={'language': 'JAVA'})
+
+
+# InferBear falls back to java if no valid language is given
+InferBearFallbackLanguageTest = verify_local_bear(
+    InferBear, valid_files=(good_java_file,), invalid_files=(bad_java_file,),
+    tempfile_kwargs={"suffix": ".java"}, settings={'language': 'bullshit'})
+
 
-InferBearTest = verify_local_bear(InferBear,
-                                  valid_files=(good_file,),
-                                  invalid_files=(bad_file,),
-                                  tempfile_kwargs={"suffix": ".java"})
+InferBearObjCTest = verify_local_bear(
+    InferBear, valid_files=(good_objc_file,), invalid_files=(bad_objc_file,),
+    tempfile_kwargs={"suffix": ".m"}, settings={'language': 'objective-c'})
```

Comment:
You can enable SSH on one of the Circle builds and use that as your linux machine.